apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryClientImpl.RoutineCrud.cs

```diff
@@ -0,0 +1,273 @@
+        {
+            GaxPreconditions.CheckNotNull(routineReference, nameof(routineReference));
+
+            var request = Service.Routines.Get(routineReference.ProjectId, routineReference.DatasetId, routineReference.RoutineId);
+            options?.ModifyRequest(request);
+            RetryHandler.MarkAsRetriable(request);
+            return request;
+        }
+
+        private ListRequest CreateListRoutinesRequest(DatasetReference datasetReference, ListRoutinesOptions options)
+        {
+            var request = Service.Routines.List(datasetReference.ProjectId, datasetReference.DatasetId);
+            options?.ModifyRequest(request);
+            RetryHandler.MarkAsRetriable(request);
+            return request;
+        }
+
+        private InsertRequest CreateInsertRoutineRequest(RoutineReference routineReference, Routine resource, CreateRoutineOptions options)
+        {
+            CheckResourceReference(routineReference, resource);
+
+            resource.RoutineReference ??= routineReference;
+
+            var request = Service.Routines.Insert(resource, routineReference.ProjectId, routineReference.DatasetId);
+            options?.ModifyRequest(request);
+            return request;
+        }
+
+        private DeleteRequest CreateDeleteRoutineRequest(RoutineReference routineReference, DeleteRoutineOptions options)
+        {
+            GaxPreconditions.CheckNotNull(routineReference, nameof(routineReference));
+            var request = Service.Routines.Delete(routineReference.ProjectId, routineReference.DatasetId, routineReference.RoutineId);
+            options?.ModifyRequest(request);
+            return request;
+        }
+
+        private UpdateRequest CreateUpdateRoutineRequest(RoutineReference routineReference, Routine resource, UpdateRoutineOptions options)
+        {
+            CheckResourceReference(routineReference, resource);
+            var request = Service.Routines.Update(resource, routineReference.ProjectId, routineReference.DatasetId, routineReference.RoutineId);
+            options?.ModifyRequest(request);
+            RetryIfETagPresent(request, resource);
+            return request;
+        }
+
+        /// <summary>
+        /// Overrides values in <paramref name="existing"/> with all set values in <paramref name="mutations"/>.
+        /// </summary>
+        /// <param name="existing">The resource that will be modified. Values in this resource will be overriden
+        /// by set values on <paramref name="mutations"/>.</param>
+        /// <param name="mutations">The resource containing the changes that will be applied to <paramref name="existing"/>.</param>
+        /// <remarks>This modifies <paramref name="existing"/>. This method only exists to provide Patch support and should
+        /// only be used with a fresh <paramref name="existing"/> resource that won't be exposed outside of the patch operation.</remarks>
+        private static void Patch(Routine existing, Routine mutations)
```

Comment:
Presumably this method goes away when there's a real patch API? I hope so - I don't like the idea of trying to keep this in sync.

Reply:
Yes, I'm weary of this as well. Wondered if we should push back and not offer patch functionality until the API does so.